variants/github_actions_ci/workflows/ci.yml

```diff
@@ -0,0 +1,157 @@
+      #   image: elasticsearch:latest
+      #   env:
+      #     cluster.name: whats-on-test-cluster
+      #     discovery.type: single-node
+      #     ES_JAVA_OPTS: "-Xms1g -Xmx1g"
+      #   ports:
+      #     - 9200:9200
+      #     - 9300:9300
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+      - name: Install Required OS Packages
+        run: sudo apt-get -yqq install libpq-dev
+      - name: Set up Ruby (version given by .ruby-version)
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+      - run: bundle exec rubocop
+      - run: bundle exec brakeman --run-all-checks --exit-on-warn --format plain .
+      - run: bundle exec bundle audit --update
+      - run: bin/setup
+      - run: bundle exec rspec spec --format progress
+  deploy_to_ec2_via_ssm_and_oidc:
+    name: Deploy with Capistrano over AWS SSM, authenticated by Github OIDC
+    runs-on: ubuntu-latest
+    needs:
+      - ruby_based_checks
+      - js_based_checks
+    timeout-minutes: 15
+    # These permissions are needed to interact with GitHub's OIDC Token endpoint.
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      # Defines the variables needed for future steps, based off of the
+      # github.ref, indicating which environment we are deploying into.
+      - name: Set variables
+        id: variables
+        run: |
+          if [[ ${{ github.ref }} == 'refs/heads/production' ]]; then
+            echo ::set-output name=ENVIRONMENT::production
+            echo ::set-output name=AWS_ROLE_TO_ASSUME::ClientNameProductionGHARole
+          elif [[ ${{ github.ref }} == 'refs/heads/main' ]]; then
+            echo ::set-output name=ENVIRONMENT::staging
+            echo ::set-output name=AWS_ROLE_TO_ASSUME::ClientNameStagingGHARole
+          fi
+      - uses: actions/checkout@v2
+      - name: Set up Ruby (version given by .ruby-version)
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+      - name: Configure AWS credentials using OpenID Connect to AssumeRole
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-session-name: GitHubActionsDeployment
+          role-to-assume: arn:aws:iam::898811473177:role/${{ steps.variables.outputs.AWS_ROLE_TO_ASSUME }}
+          aws-region: ap-southeast-2
+      - name: Install SSH key for production
+        uses: shimataro/ssh-key-action@v2
+        if: github.ref == 'refs/heads/production'
+        with:
+          key: ${{ secrets.PRODUCTION_SSH_PRIVATE_KEY }}
+          known_hosts: unnecessary
+      - name: Install SSH key for staging
+        uses: shimataro/ssh-key-action@v2
+        if: github.ref == 'refs/heads/main'
+        with:
+          key: ${{ secrets.STAGING_SSH_PRIVATE_KEY }}
+          known_hosts: unnecessary
+      - name: Deploy application
+        run: bundle exec cap ${{ steps.variables.outputs.ENVIRONMENT }} deploy
+  deploy_to_heroku:
```

Comment:
can these different deploy strategies be erb based on the other variants?

Reply:
Github Actions YAML isn't filtered through anything like ERB so I don't _think_ so.

Reply:
i mean in the write time from rails template to actual rails project

Reply:
Ah sorry, yep that's easy, just add .tt extension

Reply:
At this moment I don't _think_ we need to make this into a template.